src/python/grapl_analyzerlib/grapl_analyzerlib/viewable.py

```diff
@@ -170,6 +201,47 @@ def deserialize_edge(
         return edge_view
 
 
+def traverse_view_iter(
+    root_v: "Viewable", visited: Optional[Set["Viewable"]] = None
+) -> Iterator["Viewable"]:
+    if visited is None:
+        visited = set()
+
+    if root_v in visited:
+        return
+    yield root_v
+    visited.add(root_v)
+
+    for predicate in root_v.predicates.values():
+        if not predicate:
+            continue
+
+        if not is_edge(predicate):
+            continue
+
+        predicate = make_iter(predicate)
+
+        for pred in predicate:
+            for i_in in traverse_view_iter(pred, visited):
+                yield i_in
+
+
+def is_edge(v):
+    if isinstance(v, Viewable):
+        return True
+    if v and isinstance(v, list) and is_edge(v[0]):
+        return True
+    return False
+
+
+def make_iter(nl):
```

Comment:
nl: OneOrMany[BaseView] i assume?

(less for correct typing, more for developer sanity - because i'm lost @ what this represents)

Reply:
It takes a Union[NoneType, T, Iterator[T]] and returns an Iterator[T]